samples/MQTTSN/src/paho_iot_hub_telemetry_example_qos_0.c

```diff
@@ -0,0 +1,265 @@
+		rc = transport_sendPacketBuffer(host, port, buf, len)))
+	{
+		printf("Failed to send Connect packet to the Gateway, return code %d\n", rc);
+		return rc;
+	}
+	
+	// Wait for CONNACK from the MQTTSN Gateway
+	if (MQTTSNPacket_read(buf, buflen, transport_getdata) == MQTTSN_CONNACK)
+	{
+		int connack_rc = -1;
+
+		if (MQTTSNDeserialize_connack(&connack_rc, buf, buflen) != 1 || connack_rc != 0)
+		{
+			printf("Failed to receive Connect ACK packet, return code %d\nExiting...\n", connack_rc);
+			goto exit;
+		}
+		else 
+			printf("CONNACK rc %d\n", connack_rc);
+	}
+	else
+	{
+		printf("Failed to connect to the Gateway\nExiting...\n");
+		goto exit;
+	}
+
+	// REGISTER topic name with the MQTTSN Gateway
+	printf("Registering topic %s\n", topicname);
+	int packetid = 1;
+	topicstr.cstring = topicname;
+	topicstr.lenstring.len = strlen(topicname);
+	len = MQTTSNSerialize_register(buf, buflen, 0, packetid, &topicstr);
+	
+	if (az_failed(
+		rc = transport_sendPacketBuffer(host, port, buf, len)))
+	{
+		printf("Failed to send Register packet to the Gateway, return code %d\n", rc);
+		return rc;
+	}
+
+	// Wait for REGACK from the MQTTSN Gateway
+	if (MQTTSNPacket_read(buf, buflen, transport_getdata) == MQTTSN_REGACK) 	
+	{
+		unsigned short submsgid;
+		unsigned char returncode;
+
+		rc = MQTTSNDeserialize_regack(&topicid, &submsgid, &returncode, buf, buflen);
+		if (returncode != 0)
+		{
+			printf("Failed to receive Register ACK packet, return code %d\nExiting...\n", returncode); // TODO
+			goto exit;
+		}
+		else
+			printf("REGACK topic id %d\n", topicid);
+	}
+	else
+	{
+		printf("Failed to register topic with the Gateway\nExiting...\n"); // TODO
+		goto exit;	
+	}
+		
+	// Publish 5 messages
+	for (int i = 0; i < NUMBER_OF_MESSAGES; ++i)
+    {          
+		printf("Sending Message %d\n", i + 1);
+		topic.type = MQTTSN_TOPIC_TYPE_NORMAL;
+		topic.data.id = topicid;
+
+		// Build payload
+		unsigned char payload[250];
+		int payloadlen = sprintf((char*)payload,
+	"{\"d\":{\"myName\":\"IoT mbed\",\"accelX\":%0.4f,\"accelY\":%0.4f,\"accelZ\":%0.4f,\"temp\":%0.4f}}",
+		(rand() % 10) * 2.0, (rand() % 10) * 2.0, (rand() % 10) * 2.0, (rand() % 10) + 18.0); 
+
+		// PUBLISH
+		len = MQTTSNSerialize_publish(buf, buflen, 0, 0, retained, 0, topic, payload, payloadlen);
+
+		if (az_failed(
+			rc = transport_sendPacketBuffer(host, port, buf, len)))
+		{
+			printf("Failed to publish telemetry message %d, return code %d\n", i + 1, rc);
+			return rc;
+		}
+
+		printf("Published rc %d for publish length %d\n", rc, len);
+
+		// [BUG] doesnt work?
+		sleep_seconds(TELEMETRY_SEND_INTERVAL); // Publish a message every second
+	}
+
+	// DISCONNECT the client
+	printf("Disconnecting\n");
+	len = MQTTSNSerialize_disconnect(buf, buflen, 0);
+
+	if (az_failed(
+		rc = transport_sendPacketBuffer(host, port, buf, len)))
+	{
+		printf("Failed to send Disconnect packet to the Gateway, return code %d\n", rc);
+		return rc;
+	}
+
+	printf("Disconnected.\n");
+
+exit:

```

Comment:
While the error case of `goto`'s is great, I think we have guidance against using them in general.

Reply:
@ericwol-msft

Reply:
I really don't have issues with goto's to func exit points